### PR TITLE
Fixes P5-67: Hardware test

### PR DESCRIPTION
**Pull request by frederikspang**

### CODE REVIEW

File: report/4.Design/HardwareTest.tex [hunk 1]
```diff
@@ -0,0 +1,23 @@
+\section{Hardware testing}
+To ensure the best possible results, it is important to ensure the quality and precision of the chosen hardware to be at it's highest.
```
Comment (thomaslhansen) [inline report/4.Design/HardwareTest.tex]: Hvad definere "best possible"
Ungå at bruge "it's"

File: report/4.Design/HardwareTest.tex [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Therefore, a hardware test has been performed, over the motors, camera and laser, checking their precision, and quality.
```
Comment (thomaslhansen) [inline report/4.Design/HardwareTest.tex]: Men vi har jo ikke kontrolleret kamera og laser.

File: report/4.Design/HardwareTest.tex [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+\subsection*{Motors}
+To ensure motor capabilities, the testing has to try out combinations of precision, and to what extend the precision is reached.
```
Comment (thomaslhansen) [inline report/4.Design/HardwareTest.tex]: combinations of precision?

File: report/4.Design/HardwareTest.tex [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+Therefore a setup combining an angle, and a velocity ranging 0-100 is setup as seen in appendix \ref{appendix:motortest}.
+
+The terms: Stable, Small Correction and Huge Correction is used, to specify the stopping at the angle, and to elaborate on, not only precision, but the amount of correction needed, to stop at the exact angle specified.
+
+The tests goal was to start at a zero degree angle, speed up to the specified velocity, and stop when reaching the target angle; Just in case the motor would go further, it would automatically speed up, in the opposite direction, until reaching its goal, hence the correction.
```
Comment (thomaslhansen) [inline report/4.Design/HardwareTest.tex]: the test goal was to start... første gang du skriver i datid.

File: report/4.Design/HardwareTest.tex [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+As the motors are used to adjust the target into the center of the camera-position, a certain precision at a certain speed is required.
+This means that the motors used, and the movement speed should be able to stop at a certain angle, from that speed, or have a built-in correction as one of the weaknesses of the tests is, that the movement speed is either 0 or x, and not a gradient speed, which it may or may not bee in production.
+
+The expected result and the actual result of the test are not as surprising: The higher the speed, the higher the correction.
+However, it does not appear to be affected by the angle to which it has to aim, which is also observed in the testing; The zero to target speed is rather fast.
+
+In conclusion, in case we want to achieve the highest aim-speed as possible, we should take action upon the correction at higher speeds, and move at lower speeds when the motor approaches the target, to reduce the braking distance, thus avoiding corrections.
```
Comment (thomaslhansen) [inline report/4.Design/HardwareTest.tex]: Du bør nok tilføje noget med at vi har udvalgt de motorer der har den bedste præcision i gennemsnit ved forskellige hastigheder. Altså at vi har valgt at gå med de motor der var mest stabile.

File: report/4.Design/HardwareTest.tex [hunk 6]
```diff
@@ -0,0 +1,24 @@
+\section{Hardware testing}
+To ensure the most accurate results, it is important to ensure the quality and precision of the chosen hardware to be at its highest.
+Therefore, a hardware test has been performed on the motors, checking their precision and quality.
+
+\subsection*{Motors}
+To ensure motor capabilities, the testing has to try out combinations of target angles and movement speed, and to what extend precision is reached, considering the braking distance.
+Therefore a setup combining an angle, and a velocity ranging 0-100 is setup as seen in appendix \ref{appendix:motortest}.
```
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]:
```suggestion
Therefore a setup combining an angle, and a velocity ranging 0-100 is setup as seen in Appendix~\ref{appendix:motortest}.
```

File: report/4.Design/HardwareTest.tex [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+The terms: Stable, Small Correction and Huge Correction is used, to specify the stopping at the angle, and to elaborate on, not only precision, but the amount of correction needed, to stop at the exact angle specified.
```
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]: are used

File: report/4.Design/HardwareTest.tex [hunk 8]
```diff
@@ -0,0 +1,24 @@
+\section{Hardware testing}
+To ensure the most accurate results, it is important to ensure the quality and precision of the chosen hardware to be at its highest.
```
[the remark this answers is not in the document]
Comment (nillas12) [reply report/4.Design/HardwareTest.tex]: Maybe change to something akin to: "to ensure the accuracy of the device"...

File: report/4.Design/HardwareTest.tex [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+Therefore, a hardware test has been performed on the motors, checking their precision and quality.
```
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: Think this should be "Therefore, a test of the motors has been conducted, ...." or something

File: report/4.Design/HardwareTest.tex [hunk 10]
```diff
@@ -0,0 +1,24 @@
+\section{Hardware testing}
```
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: I think in general this section needs to be streamlined in terms of what tense it uses. Because I actually think that past tense is wrong.

File: report/4.Design/HardwareTest.tex [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+To ensure the most accurate results, it is important to ensure the quality and precision of the chosen hardware to be at its highest.
+Therefore, a hardware test has been performed on the motors, checking their precision and quality.
+
+\subsection*{Motors}
+To ensure motor capabilities, the testing has to try out combinations of target angles and movement speed, and to what extend precision is reached, considering the braking distance.
```
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: extend -> extent
also maybe just change the final part to something like: and to determine how precise the motors are. Dunno, I just dont really like that final part too much.

File: report/Appendix/motor_test.tex [hunk 12]
```diff
@@ -0,0 +1,120 @@
+\subsection{Motor-test}\label{appendix:motortest}
```
[the remark this answers is not in the document]
Comment (nillas12) [reply report/Appendix/motor_test.tex]: yaasas

[the remark this answers is not in the document]
Comment (nillas12) [reply report/Appendix/motor_test.tex]: Maybe actually in the actual section itself

File: report/4.Design/HardwareTest.tex (hunk 11 above)
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]:
```suggestion
To ensure the capabilities of the motors, the tests have to try out combinations of target angles and movement speed, and to what extend precision is reached, considering the braking distance.
```

den er stadig lidt skrald, hjælp @nillas12

File: report/4.Design/HardwareTest.tex [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+The expected result and the actual result of the test are not as surprising: The higher the speed, the higher the correction.
+However, it does not appear to be affected by the angle to which it has to aim, which is also observed in the testing; The zero to target speed is rather fast.
+
+In conclusion, in case we want to achieve the highest aim-speed as possible, we should take action upon the correction at higher speeds and move at lower speeds when the motor approaches the target, to reduce the braking distance, thus avoiding corrections.
+The results of the tests are therefore used to not only choose which motors to use, but also giving a pointer to an optimal movement speed.
+
+\subsection*{Camera}
+Due to the use of the camera, the actual quality of the footage is secondary.
+In this case, the camera footage will be scaled down to a lower resolution and frame rate, rendering the actual camera quality irrelevant to a certain extend.
```
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: extend -> extent

File: report/4.Design/HardwareTest.tex (hunk 7 above)
Comment (Moesgaarda) [reply report/4.Design/HardwareTest.tex]: btw skal de stå med store?

Comment (Moesgaarda) [reply report/4.Design/HardwareTest.tex]:
```suggestion
The terms stable, small correction and huge correction are used to specify the stopping at the angle, and to elaborate on, not only precision, but the amount of correction needed, to stop at the exact angle specified.
```

jeg forstår ikke "to specity the stopping at the angle" sætningen, er jeg retarderet???????

File: report/4.Design/HardwareTest.tex [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+To ensure the most accurate results, it is important to ensure the quality and precision of the chosen hardware to be at its highest.
+Therefore, a hardware test has been performed on the motors, checking their precision and quality.
+
+\subsection*{Motors}
+To ensure motor capabilities, the testing has to try out combinations of target angles and movement speed, and to what extend precision is reached, considering the braking distance.
+Therefore a setup combining an angle, and a velocity ranging 0-100 is setup as seen in appendix \ref{appendix:motortest}.
+
+The terms: Stable, Small Correction and Huge Correction is used, to specify the stopping at the angle, and to elaborate on, not only precision, but the amount of correction needed, to stop at the exact angle specified.
+
+The tests goal is to start at a zero degree angle, speed up to the specified velocity, and stop when reaching the target angle; Just in case the motor will go further, it will automatically speed up, in the opposite direction, until reaching its goal, hence the correction.
```
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]:
```suggestion
The tests goal is to start at a zero degree angle, speed up to the specified velocity, and stop when reaching the target angle; just in case the motor will go further, it will automatically speed up, in the opposite direction, until reaching its goal, hence the correction.
```

Kun stort efter semikolon hvis det er et proper noun.

File: report/4.Design/HardwareTest.tex [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+As the motors are used to adjust the target into the center of the camera-position, a certain precision at a certain speed is required.
```
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]: vil gerne af med den ene certain

File: report/4.Design/HardwareTest.tex (hunk 14 above)
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: The tests? I am genuinely not sure if it should be tests, test's, tests'. But for some reason tests ring wrong to me.

Comment (nillas12) [reply report/4.Design/HardwareTest.tex]: Also, while this is technically ok, I do think this sentence is a bit long (thats what she said, although it was not in reference to a sentence, but rather to ma dick)

File: report/4.Design/HardwareTest.tex [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+This means that the motors used, and the movement speed should be able to stop at a certain angle, from that speed, or have a built-in correction as one of the weaknesses of the tests is, that the movement speed is either 0 or x, and not a gradient speed, which it may or may not bee in production.
```
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]: either 0 or x?

Gradient speed? Varying?

og der skal helst ikke være bier i produktionen

File: report/4.Design/HardwareTest.tex [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+
+The expected result and the actual result of the test are not as surprising: The higher the speed, the higher the correction.
+However, it does not appear to be affected by the angle to which it has to aim, which is also observed in the testing; The zero to target speed is rather fast.
```
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]: ikke stort efter semikolon

File: report/4.Design/HardwareTest.tex [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+
+In conclusion, in case we want to achieve the highest aim-speed as possible, we should take action upon the correction at higher speeds and move at lower speeds when the motor approaches the target, to reduce the braking distance, thus avoiding corrections.
```
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]: undgår vi helt corrections?

File: report/4.Design/HardwareTest.tex (hunk 13 above)
Comment (Moesgaarda) [inline report/4.Design/HardwareTest.tex]: er den her sektion færdig? synes at der mangler noget dybde

File: report/4.Design/HardwareTest.tex (hunk 15 above)
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: I don't like the final part of this sentence

File: report/Appendix/motor_test.tex [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+
+\begin{longtable}[c]{|ccccccc|}
+\hline
+Id & Model & Speed & Angle & Angle measued +-1 & Difference & Comment \\\hline
+\endfirsthead
+%
+\multicolumn{7}{c}%
+{{\bfseries Table \thetable\ continued from previous page}} \\\hline
+Id & Model & Speed & Angle & Angle measued +-1 & Difference & Comment \\\hline
```
Comment (Moesgaarda) [inline report/Appendix/motor_test.tex]: measured

File: report/Appendix/motor_test.tex [hunk 20]
```diff
@@ -0,0 +1,120 @@
+\subsection{Motor-test}\label{appendix:motortest}
+
+\begin{longtable}[c]{|ccccccc|}
+\hline
+Id & Model & Speed & Angle & Angle measued +-1 & Difference & Comment \\\hline
```
Comment (Moesgaarda) [inline report/Appendix/motor_test.tex]: measured

File: report/4.Design/HardwareTest.tex (hunk 10 above)
Comment (nillas12) [reply report/4.Design/HardwareTest.tex]: Also I know my comments are not as good as Daniels but I am in a creative rut, and can only detect things I don't like

File: report/4.Design/HardwareTest.tex (hunk 18 above)
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: This sentence reads a bit strange. It might actually be ok, but it just gets a bit long I think

File: report/4.Design/HardwareTest.tex [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+The results of the tests are therefore used to not only choose which motors to use, but also giving a pointer to an optimal movement speed.
```
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: I don't like the use of "giving a pointer".

File: report/4.Design/HardwareTest.tex [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+Therefore a setup combining an angle, and a velocity ranging 0-100 is setup as seen in appendix \ref{appendix:motortest}.
+
+The terms: Stable, Small Correction and Huge Correction is used, to specify the stopping at the angle, and to elaborate on, not only precision, but the amount of correction needed, to stop at the exact angle specified.
+
+The tests goal is to start at a zero degree angle, speed up to the specified velocity, and stop when reaching the target angle; Just in case the motor will go further, it will automatically speed up, in the opposite direction, until reaching its goal, hence the correction.
+
+As the motors are used to adjust the target into the center of the camera-position, a certain precision at a certain speed is required.
+This means that the motors used, and the movement speed should be able to stop at a certain angle, from that speed, or have a built-in correction as one of the weaknesses of the tests is, that the movement speed is either 0 or x, and not a gradient speed, which it may or may not bee in production.
+
+The expected result and the actual result of the test are not as surprising: The higher the speed, the higher the correction.
```
Comment (nillas12) [inline report/4.Design/HardwareTest.tex]: Would it not be "...correction required"? Or do I misunderstand

File: report/4.Design/HardwareTest.tex (hunk 13 above)
Comment (frederikspang) [reply report/4.Design/HardwareTest.tex]: Det var dét eller ud med den.. Problemet er at der intet er at sige her.

File: report/4.Design/HardwareTest.tex (hunk 18 above)
Comment (frederikspang) [reply report/4.Design/HardwareTest.tex]: Hvis vi kommer langt nok ned i bremseafstand gør vi, jf. bilaget ("Stable")

File: report/4.Design/HardwareTest.tex (hunk 6 above)
[the remark this answers is not in the document]
Comment (frederikspang) [reply report/4.Design/HardwareTest.tex]: 0 er stilstand, og også en "velocity", vel?

File: report/4.Design/HardwareTest.tex [hunk 23]
```diff
@@ -0,0 +1,24 @@
+\section{Hardware testing}
+To ensure the most accurate of the device, it is important to ensure the quality and precision of the chosen hardware to be at its highest.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: "To ensure the most accurate of the device"

"To ensure the most accuracy of the device"

File: report/4.Design/HardwareTest.tex [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+Therefore, a test of the motors has been conducted, checking their precision and quality.
+
+\subsection*{Motors}
+To ensure the capabilities of the motors, the tests have to try out combinations of target angles and movement speed, and to what extent precision is reached, considering the braking distance.
+Therefore a setup combining an angle, and a velocity ranging 0-100 is setup as seen in Appendix~\ref{appendix:motortest}.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: "Ranging 0-100" giver ikke mening for mig... (eller jo men det lyder forkert)
"ranging from 0 to 100"
eller "in range 0 to 100" (eller 0-100)

File: report/4.Design/HardwareTest.tex [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+
+The terms stable, small correction and huge correction are used to specify the stopping at the angle, and to elaborate on, not only precision, but the amount of correction needed, to stop at the exact angle specified.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: hvis du bruger et term skal det på en eller anden måde have highlighting på at det er termet.. så enten italic eller citationstegn or something. vi bruger vist generelt italic i rapporten.

File: report/4.Design/HardwareTest.tex [hunk 26]
```diff
@@ -0,0 +1,24 @@
...
+
+The tests' goal is to start at a zero degree angle, speed up to the specified velocity, and stop when reaching the target angle; just in case the motor will go further, it will automatically speed up, in the opposite direction, until reaching its goal, hence the correction.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: "the goal of the tests" synes jeg lyder bedre.

Comment (C0DK) [reply report/4.Design/HardwareTest.tex]: "speed up" er vel forkert i dine tests. det er vel "vi kører med hastighed X eller ikke med hastighed X". så "set speed" eller lign

File: report/4.Design/HardwareTest.tex [hunk 27]
```diff
@@ -0,0 +1,24 @@
...
+
+As the motors are used to adjust the target into the center of the camera-position, a certain precision at a given speed is required.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: as the motors are used to aim, precision is a high valued quality.

File: report/4.Design/HardwareTest.tex [hunk 28]
```diff
@@ -0,0 +1,24 @@
...
+This means that the motors used, and the movement speed should be able to stop at a certain angle, from that speed, or have a built-in correction as one of the weaknesses of the tests is, that the movement speed is either 0 or x, and not a varying velocity, which it may or may not be in production.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: "this means that the motors used, should be able to stop at a certain angle, when running at speed x. " er bedre. men stadig en mærkelig del-sætning.

"or have a built-in correction as one of the weaknesses of the tests is, that the movement speed is either 0 or x, and not a varying velocity, which it may or may not be in production."
virker som volapyk. men split evt hele paragraffen op i flere sætninger.

File: report/4.Design/HardwareTest.tex [hunk 29]
```diff
@@ -0,0 +1,24 @@
...
+
+The expected result and the actual result of the test are not as surprising: The higher the speed, the more correction required.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: ret til: "the more correction *is* required"


"The expected result and the actual result of the test are not as surprising" -> betyder det at relationen mellem de 2 ikke er overraskende? eller at det endelige resultat ikke er?

File: report/4.Design/HardwareTest.tex [hunk 30]
```diff
@@ -0,0 +1,24 @@
...
+However, it does not appear to be affected by the angle to which it has to aim, which is also observed in the testing; the zero to target speed is rather fast.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: "the zero to target speed is rather fast."

ret til:
"Acceleration to the target speed happens rather fast"

File: report/4.Design/HardwareTest.tex [hunk 31]
```diff
@@ -0,0 +1,24 @@
...
+
+In conclusion, in case we want to achieve the highest aim-speed as possible, we should take action upon the correction at higher speeds and move at lower speeds when the motor approaches the target, to reduce the braking distance, thus avoiding corrections.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: "In conclusion, in case we want to achieve the highest aim-speed as possible" ->
"In conclusion, in case we want to achieve the highest possible aim-speed"

File: report/4.Design/HardwareTest.tex [hunk 32]
```diff
@@ -0,0 +1,24 @@
...
+The results of the tests are therefore used to not only choose which motors to use, but also gives a hint towards what the optimal movement speed is.
+
+\subsection*{Camera}
+Due to the use of the camera, the actual quality of the footage is secondary.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: jeg ville måske skrive noget andet og sige at det var vigtigt at vi nogenlunde klart kan se objektet på en video feed. Dette testede vi kort ved at tage nogle videoer hvor vi bevægede en bold i forskellige hastigheder, og tjekkede hvor sløret billedet blev. Den reelle opløsning er ikke vigtigt, da vi ikke skal kunne se motiver klart, men hellere fokuserer på eksekveringshastigheden. mhtFramerate er kameraet ikke en bottleneck, men nærmere vores software. Derdover bliver opløsningen nok nedskaleret.

File: report/4.Design/HardwareTest.tex [hunk 33]
```diff
@@ -0,0 +1,24 @@
+\section{Hardware testing}
+To ensure the most accurate of the device, it is important to ensure the quality and precision of the chosen hardware to be at its highest.
+Therefore, a test of the motors has been conducted, checking their precision and quality.
```
Comment (C0DK) [inline report/4.Design/HardwareTest.tex]: a test of the motors and the chosen camera
